aiohttp/_http_parser.pyx

```diff
@@ -378,42 +379,35 @@ cdef class HttpParser:
         self._limit = limit
 
     cdef _process_header(self):
-        if self._raw_name:
-            raw_name = bytes(self._raw_name)
-            raw_value = bytes(self._raw_value)
-
-            name = find_header(raw_name)
-            value = raw_value.decode('utf-8', 'surrogateescape')
+        cdef str value
+        if self._raw_name is not EMPTY_BYTES:
+            name = find_header(self._raw_name)
+            value = self._raw_value.decode('utf-8', 'surrogateescape')
 
-            self._headers.add(name, value)
+            self._headers.append((name, value))
 
```

Comment:
Its a bit faster to build up a list of tuples and have `CIMultiDict` build from that instead of calling `.add()` over and over since we can stay in Cython without having to jump back to python code for every header since the only time we have to end up in Python is to pass the list to `CIMultiDict`